SOURCE/set_PressureTemplate.c

```diff
@@ -0,0 +1,454 @@
+        fid = fopen(model.input_file, "rb"); // Open file
+        if (!fid){
+            fprintf(stderr, "\nUnable to open file %s. I will exit here ... \n", model.input_file); exit(2);
+        }
+        fread( ph_hr, sizeof(char), nb_elems, fid);
+        fclose(fid);
+    }
+    // image properties
+    int nx_hr = 1922, nz_hr = 1922, ix, iz;
+    double dx_hr = (model.xmax - model.xmin)/(nx_hr-1.0);
+    double dz_hr = (model.zmax - model.zmin)/(nz_hr-1.0);
+    double dstx, dstz;
+    
+    // Loop on particles
+    for( np=0; np<particles->Nb_part; np++ ) {
+        
+        // Standard initialisation of particles
+        particles->Vx[np]    = -1.0*particles->x[np]*model.EpsBG;               // set initial particle velocity (unused)
+        particles->Vz[np]    =  particles->z[np]*model.EpsBG;                   // set initial particle velocity (unused)
+        particles->phase[np] = 0;                                               // same phase number everywhere
+        particles->d[np]     = 0;                                               // same grain size everywhere
+        particles->phi[np]   = 0.0;                                             // zero porosity everywhere
+        particles->T[np]     = T_init;
+        particles->P[np]     = P_init;
+        particles->noise[np] = ((double)rand() / (double)RAND_MAX) - 0.5;
+        
+        if (setup==2) {
+            // ------------------------- //
+            // Locate markers in the image files
+            // Find index of minimum/west temperature node
+            dstx = ( particles->x[np] - model.xmin );
+            ix   = ceil( dstx/dx_hr - 0.0 ) - 1;
+            // Find index of minimum/west pressure node
+            dstz = ( particles->z[np] - model.zmin );
+            iz   = ceil( dstz/dz_hr  - 0.0 ) - 1;
+            // Attribute phase
+            if (ix<0) {printf("sauceisse!!!\n"); exit(1);}
+            if (iz<0) {printf("puréee!!!\n"); exit(1);}
+            if (ix + iz*(nx_hr-1) > nb_elems) {printf("puréee!!!\n"); exit(1);}
+    //        printf("%d\n", (int)ph_hr[ix + iz*(nx_hr-1)]);
+            particles->phase[np] = (int)ph_hr[ix + iz*(nx_hr-1)];
+        }
+        if (setup==1) {
+            // DRAW INCLUSION
+            X  = particles->x[np]-xc;
+            Z  = particles->z[np]-zc;
+            Xn = X*cos(theta) - Z*sin(theta);
+            Zn = X*sin(theta) + Z*cos(theta);
+            if ( pow(Xn/la,2) + pow(Zn/sa,2) - 1 < 0 ) particles->phase[np] = 1;
+        }
+        
+        // SANITY CHECK
+        if (particles->phase[np] > model.Nb_phases-1) {
+            printf("Lazy bastard! Fix your particle phase ID! \n");
+            exit(144);
+        }
+
+        // SET DEFAULT DUAL PHASE NUMBER
+        particles->dual[np] = particles->phase[np];
+
+        //--------------------------//
+    }
+
+    // Generate checkerboard
+    for ( np=0; np<particles->Nb_part; np++ ) {
+            Ax = cos( 6.0*2.0*M_PI*particles->x[np] / Lx  );
```

Comment:
Lx and Lz are used here for the first time buy they are initialised almost 100 lines before
They should be initialised before the usage

Reply:
Ok